Sources/Sentry/SentrySpan.m

```diff
@@ -33,6 +36,20 @@ - (instancetype)initWithContext:(SentrySpanContext *)context
     if (self = [super init]) {
         self.startTimestamp = [SentryDependencyContainer.sharedInstance.dateProvider date];
         _data = [[NSMutableDictionary alloc] init];
+
+        SentryCrashThread currentThread = sentrycrashthread_self();
+        _data[SPAN_DATA_THREAD_ID] = @(currentThread);
+
+        if ([NSThread isMainThread]) {
+            _data[SPAN_DATA_THREAD_NAME] = @"main";
+        } else {
+            NSString *threadName = [SentryDependencyContainer.sharedInstance.threadInspector
+                getThreadName:currentThread];
+            if (threadName.length > 0) {
+                _data[SPAN_DATA_THREAD_NAME] = threadName;
+            }
+        }
```

Comment:
Any reason we couldn't put this block of logic into SentryThreadInspector, so that the code here would just read as
```
_data[SPAN_DATA_THREAD_NAME] = SentryDependencyContainer.sharedInstance.threadInspector.threadName;
```

Reply:
ThreadInspector threadName returns an empty string when the thread name is `nil`, see https://github.com/getsentry/sentry-cocoa/blob/ae9c51b0d6e71205648e9d44d9f4d61827ccd1e6/Sources/Sentry/SentryThreadInspector.m#L211-L215

I will fix that in another PR.

Good catch, thanks @armcknight 👍

Reply:
I opened a PR https://github.com/getsentry/sentry-cocoa/pull/3361